Assets/__Scripts/MapEditor/Mapping/Selection/BoxSelectionPlacementController.cs

```diff
@@ -83,13 +118,55 @@ public override void OnPhysicsRaycast(RaycastHit hit, Vector3 transformedPoint)
             newLocalScale = new Vector3(newLocalScale.x, newLocalScaleY, newLocalScale.z);
             instantiatedContainer.transform.localScale = newLocalScale;
 
-            OverlapBox((containerBoye) =>
+            float startBeat = instantiatedContainer.transform.localPosition.z / EditorScaleController.EditorScale;
+            float endBeat = (instantiatedContainer.transform.localPosition.z + newLocalScale.z) / EditorScaleController.EditorScale;
+            if (startBeat > endBeat) (startBeat, endBeat) = (endBeat, startBeat);
+
+            SelectionController.ForEachObjectBetweenTimeByGroup(startBeat, endBeat, true, true, true, (bocc, bo) =>
             {
-                if (!alreadySelected.Contains(containerBoye.objectData) && selected.Add(containerBoye.objectData))
+                if (!SelectedTypes.Contains(bo.beatmapType)) return; // Must be a type we can select
+
+                var left = instantiatedContainer.transform.localPosition.x + instantiatedContainer.transform.localScale.x;
+                var right = instantiatedContainer.transform.localPosition.x;
+                if (right < left) (left, right) = (right, left);
+
+                var top = instantiatedContainer.transform.localPosition.y + instantiatedContainer.transform.localScale.y;
+                var bottom = instantiatedContainer.transform.localPosition.y;
+                if (top < bottom) (top, bottom) = (bottom, top);
+
+                if (bo is BeatmapNote note)
```

Comment:
With all of these Position and Center variables, I almost want to consider abstracting it all into `BeatmapObjectBounds` class, and having an abstract method/property in `BeatmapObject` which constructs it from the map data.